content/en/blog/2022/frontend-overhaul/index.md

```diff
@@ -0,0 +1,182 @@
+5. Ruby
+6. Python
+7. Node.js
+8. Rust
+9. Elixir
+
+Every microservice has a specific goal and can communicate with others by using
+a global GRPC definition. Persistent information is saved into a Postgres
+database and there are outbound services that connect with third-party services
+to trigger events (such as confirmation emails). All of the microservices,
+including the front end, are connected to the same Open Telemetry collector
+instance, which uses Jaeger as one of the data stores for the traces and spans.
+
+![Open Telemetry Demo System Diagram](system-diagram.png)
+
+The front end was constituted by a Golang SSR application which sent the
+complete HTML to the browser client to be displayed. Each request and form call
+was redirecting the user back to the server so the new piece of information was
+shown.
+
+## Web App Styling Improvements, Theme Updates, and User Experience Redesign
+
+Before starting the development process, the frontend application wasn’t
+matching the theme that OpenTelemetry had been using in terms of colors,
+products, and overall user experience. In addition, the demo lacked a real
+frontend (browser side) application as the current implementation was a Go SSR
+application.
+
+![Open Telemetry Demo Old Frontend](old-design.png)
+
+The first task at hand was to bring the demo to the modern age by updating the
+design, color schemes, and user experience. Olly Babiak walked into the fray to
+help us achieve this by creating a modernized version of the application. It
+included an improved way to display the products landing page, an updated
+product details page, a mini cart, and a fully compatible mobile version of the
+application.
+
+![Open Telemetry Demo New Frontend](new-design.png)
+
+Now we had an application design that would match the rest of the OpenTelemetry
+themes and colors and look more like the OpenTelemetry.io website.
+
+## Frontend Application Architecture Overhaul
+
+We worked on an initial proposal that included the following bullet points:
+
+- Framework and tooling (Scaffolding, I/O, styling, UI library)
+- Code Architecture and structure (Directories, coding patterns)
+- Instrumentation
+- Deployment & Distribution
+- Testing (E2E, unit test)
+
+This proposal was presented to the OpenTelemetry Demo SIG duringone of the
+weekly Monday meetings and we were given the green light to move ahead. As part
+of the changes, we decided to use Next.js to not only work as the primary
+frontend application but also to work as an aggregation layer between the
+frontend and the GRPC backend services.
+
+![New Frontend Data Flow](data-flow.png)
+
+As you can see in the Diagram, the application has two major connectivity
+points, one coming from the browser side (REST) to connect to the Next.js
+aggregation layer and the other from the aggregation layer to the backend
+services (GRPC).
+
+## OpenTelemetry Instrumentation
+
+The next big thing we worked on was having a way to instrument both sides of the
```

Comment:
Rewording:
“… was a way to instrument both sides of the Next.js app. To do this we had to connect the app twice to the same collector used by all the microservices.”